### ISSUE TITLE
Put brief (<65 char) title here!

### ISSUE THREAD

**Issue by cacraigucar** [ESCOMP/CAM#692]
### What is the feature/what would you like to discuss?

Compset naming conventions need to be expanded for the upcoming CAM7

Proposal:

F{W}{C}{LT|MT}{scenario}{Slim}

F will include new prognostic greenhouse gases (This will replace the specified chemistry option)
W if is a WACCM run
C if is a full chemistry run
LT for low top or MT for mid top 
    LT = 58 levels currently, initial condition files, tau_0_ubc=false + other tunings if needed later
    MT = 93 levels currently, initial condition files, tau_0_ubc=true + other tunings if needed later
scenario = HIST, 1850, 2000Climo, etc.
Slim - Indicates SLIM Land model if appropriate

The compsets F2000Climo, FHIST and F1850 will be deprecated as the 32 level will not be supported

Examples:
FMTHIST will be mid top historical
FCLTHIST will be full chemistry low top historical
FWC2000Climo will be WACCM with chemisty in a 2000 climo scenario
FLT2000ClimoSlim will be low top 2000 climo scenario with CTSM SLIM model

### Is there anyone in particular you want to be part of this conversation?

@cecilehannay @JulioTBacmeister

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Yes, but I will need some help

**Comment by jedwards4b** [comment]
Will LT or MT always be required?  If one of these is to be the default perhaps we only need to specify the other? So
if LT is default then we would have FHIST and FMTHIST.

**Comment by fvitt** [comment]
@jedwards4b I think, for the time being "LT" indicates cam_dev physics with 58 vertical layers.  The current default is cam6 physics with 32 vertical layers.

**Comment by jedwards4b** [comment]
But will we continue to support a 32 layer model?  If so perhaps it should be LT?

**Comment by cacraigucar** [comment]
I asked @JulioTBacmeister the question yesterday of whether or not we will be supporting 32 layer CAM in CAM7 and his answer was no.  We will be supporting both 58 and 93 which is why the proposal is to have the "LT" and "MT" designators in the compset names.

**Comment by swrneale** [comment]
I kind of agree with Jim. If we will ultimately view L93 as the default/work horse then we don't need a HT specifier as we want continuity of people running the default model FHIST/BHIST etc. compset. It' tricky though as I suspect most people will be making L58 comparisons with CAM6 and not L93.

**Comment by adamrher** [comment]
I agree with Rich's point about the continuity of naming conventions, so that for e.g., FHIST/BHIST points to whatever we consider the new CAM7 workhorse (-phys cam_dev -nlev 58/93).

Likewise, this permits folks to easily revert to older workhorse/physics by running FHIST and setting the -phys in CAM_CONFIG_OPTS. The convention has been that -phys determines the vertical grid (in bld/configure), which is then further refined by the -chem waccm flag to select a high top grid.

So just to be clear, @jedwards4b @cacraigucar I think that @JulioTBacmeister is saying we won't support the 32 level for -phys cam_dev. The 32 level model should still run by setting -phys cam6.

**Comment by JulioTBacmeister** [comment]
We will no longer support 32 level configurations in CAM7. I continue to favor using both 'FLT' and 'FMT' in the compset name.  However, I need to learn more about Adam's comment above.  

The reason I favor FLT/FMT is that while FMT will be the new workhorse configuration, FMT is a substantial departure from  previous F compsets in most respects - including not only model top height and vertical resolution but also treatment of greenhouse gases and gravity waves.  FLT will be closer to previous F compsets in model top height and possibly GW parameterization.  

I am afraid using "F" for the new 85km/L93 configuration will be very confusing since the physics differences between this and previous "F" compsets are so significant.  Especially since "FLT" will actually be closer to the previous "F" compsets in most repsects.

Howvever, using "F" for the new 45km/L58 configuration is inconsistent with previous practice and misleading as it implies 45km/L58 is somehow the new default.  In addition FLT will likely contain a significant departure in the treatment of greenhouse gases with respect to earlier "F" compsets.

I favor marking both new configurations explicitly in the compset names for clarity. It doesn't hurt to remind the user community that both configurations are a substantial departure from previous CAM configurations.

**Comment by adamrher** [comment]
Julio you make good points. But I think you are too narrowly constraining what F has meant in the model in the past. It will refer to running the latest physics package. In this case, cam_dev / CAM7. Differences in how GW is parameterized, changes in the default behavior of how the trace constituents such as aerosols interact with cloud liquid/ice or how chemical species are prescribed or interact with radiation is often changed from one version to another. I think about how much more complex CAM5 physics are compared to CAM4.

And so taken together with your arguments that "FLT" is much more similar to what we would normally just call "F", I'd vote for just calling it "F" ... that is, two compsets FHIST and FMTHIST. And that way it's clear we've moved the goal posts on our non-WACCM workhorse model -- FMTHIST -- to be a medium top model with a much better resolved stratosphere.

**Comment by ekluzek** [comment]
I just want to add a small note about SLIM. SLIM is the simple land model that you will be able to run in CESM to replace CLM. It can't work with WACCM or chemistry so there should only be a FLT variant with SLIM in the compset name. But you could have different scenarios like 1850 or HIST...

**Comment by adamrher** [comment]
Hi @ekluzek , this looks interesting. [I google'd it](https://www.cesm.ucar.edu/events/workshops/ws.2019/presentations/plenary/lague.pdf). Having personally struggled to perturb one property at a time in the land model, I think this is great. It actually sounds like a good candidate for the [simpler models compsets](https://www.cesm.ucar.edu/models/simpler-models/). We should touch base about this, but separately from this thread as its focused on the workhorse configurations that include the latest greatest CTSM and all its bells and whistles.

**Comment by ekluzek** [comment]
@adamrher thanks for your thoughts here. Yes, use of SLIM totally goes with the simple model options of CAM. If this is only about the workhorse configurations it possibly shouldn't be considered in this list. Is there also planning on how to handle the simple model compsets for CAM? You are right SLIM more naturally falls into that camp. SLIM is a little more complex than some of the other "simple model" options, so it's a little in it's own class. But since it can't do: chemistry, river or land-ice it can't be used with the most complex options of CAM either.

**Comment by dan800** [comment]
Interesting discussion above and I can see both sides. I would lean towards @JulioTBacmeister position that it does not hurt to call out the model top in the compset name. That way it's very clear in the diagnostic plot titles what is being run. At least until we admit that to be a useful model you need a resolved stratosphere and the LT is no longer supported (I can dream, right?). Then we can go back to an unambiguous F for the workhorse model.

**Comment by cacraigucar** [comment]
We have an upcoming PR #669 which utilizes the `LT` and `MT` naming convention.  Are folks okay with at least that part of this naming convention?

**Comment by swrneale** [comment]
It seems very reasonable at this point in time.

**Comment by adamrher** [comment]
fine by me
